src/backend/cdb/dispatcher/cdbdisp.c

```diff
@@ -589,6 +606,25 @@ cdbdisp_cleanupDispatcherHandle(const struct ResourceOwnerData *owner)
 	}
 }
 
+void
+cdbdisp_destroyDispatcherHandle(const struct ResourceOwnerData *owner)
+{
+	dispatcher_handle_t *curr;
+	dispatcher_handle_t *next;
+
+	next = open_dispatcher_handles;
+	while (next)
+	{
+		curr = next;
+		next = curr->next;
+
+		if (curr->owner == owner)
+		{
+			destroy_dispatcher_state_and_handle(curr);
+		}
+	}
+}
+
```

Comment:
Is it possible to extract the common parts of similar functions `cdbdisp_cleanupDispatcherHandle` and `cdbdisp_destroyDispatcherHandle` into one?